File: java/src/main/java/mx/iteso/Priviledge/Comment.java
package mx.iteso.Priviledge;

import mx.iteso.PriviledgeDecorator;
import mx.iteso.User;

public class Comment extends PriviledgeDecorator {
    User user;

    public Comment (User user){
        this.user = user;
    }

    @Override
    public String getDescription() {
        return user.getDescription() + " can Comment";
    }
}

File: csharp/decorator-practice/decorator-practice/Comment.cs
namespace decorator_practice
{
    internal class Comment : PrivilegesDecorator
    {
        private readonly User _user;


        public Comment(User userIn)
        {
            _user = userIn;
        }

        override
        public string PublishM()
        {
            return _user.PublishM();
        }

        override
        public string DeleteM()
        {
            return _user.DeleteM();
        }

        override
        public string ApproveM()
        {
            return _user.ApproveM();
        }

        override
        public string CommentM()
        {
            return "Can comment";
        }
    }
}
